src/pages/iou/request/step/IOURequestStepMerchant.tsx

```diff
@@ -71,30 +61,28 @@ function IOURequestStepMerchant({
     policyTags,
     policyCategories,
     report,
-}) {
+}: IOURequestStepMerchantProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {inputCallbackRef} = useAutoFocusInput();
     const isEditing = action === CONST.IOU.ACTION.EDIT;
 
     // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
     const isEditingSplitBill = iouType === CONST.IOU.TYPE.SPLIT && isEditing;
-    const {merchant} = ReportUtils.getTransactionDetails(isEditingSplitBill && !lodashIsEmpty(splitDraftTransaction) ? splitDraftTransaction : transaction);
+    const merchant = ReportUtils.getTransactionDetails(isEditingSplitBill && !isEmptyObject(splitDraftTransaction) ? splitDraftTransaction : transaction)?.merchant;
     const isEmptyMerchant = merchant === '' || merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
-    const isMerchantRequired = ReportUtils.isGroupPolicy(report) || _.some(transaction.participants, (participant) => Boolean(participant.isPolicyExpenseChat));
+
+    // @ts-expect-error TODO: Remove this once withFullTransactionOrNotFound (https://github.com/Expensify/App/issues/36123) is migrated to TypeScript.
+    const isMerchantRequired = ReportUtils.isGroupPolicy(report) || transaction?.participants?.some((participant) => Boolean(participant.isPolicyExpenseChat));
     const navigateBack = () => {
         Navigation.goBack(backTo);
     };
 
-    /**
-     * @param {Object} value
-     * @param {String} value.moneyRequestMerchant
-     */
     const validate = useCallback(
-        (value) => {
-            const errors = {};
+        (value: ValidationData) => {
+            const errors: ValidationData = {};
 
-            if (isMerchantRequired && _.isEmpty(value.moneyRequestMerchant)) {
+            if (isMerchantRequired && (value.moneyRequestMerchant ?? '').length <= 0) {
```

Comment:
```suggestion
            if (isMerchantRequired && (value.moneyRequestMerchant ?? '').length === 0) {
```

it will never be less than 0

Reply:
Since `value.moneyRequestMerchant` is string
```suggestion
            if (isMerchantRequired && !value.moneyRequestMerchant) {
```